rewrite-java/src/main/java/org/openrewrite/java/cleanup/UnnecessaryParentheses.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2020 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.cleanup;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.style.IntelliJ;
+import org.openrewrite.java.style.UnnecessaryParenthesesStyle;
+import org.openrewrite.java.tree.J;
+
+public class UnnecessaryParentheses extends Recipe {
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new UnnecessaryParenthesesFromCompilationUnitStyle();
+    }
+
+    private static class UnnecessaryParenthesesFromCompilationUnitStyle extends JavaIsoVisitor<ExecutionContext> {
```

Comment:
The name of this seems a bit long. How about `UnnecessaryParenthesisVisitor` ?

Reply:
It's a bit long though not meant to be called by consumers through a public api. It's following the pattern used in formatting-related recipes in the `format` package (e.g. BlankLines, Spaces, etc.)

There's a different file for `UnnecessaryParenthesisVisitor` which contains the actual `visitor` logic implementation, whereas `UnnecessaryParenthesesFromCompilationUnitStyle` is responsible for loading the `style` to be used by the actual `UnnecessaryParenthesisVisitor` implementation